core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -0,0 +1,150 @@
+        DateTime.parse(row.getF.get(3).getV.toString),
+        DateTime.parse(row.getF.get(3).getV.toString).plusDays(1).minusSeconds(1))
+    }
+    val dailySpendAggregation = SpendReportingAggregation(
+      SpendReportingAggregationKey("total"), dailySpend
+    )
+
+    val costRollup = rows.map { row =>
+      BigDecimal(row.getF.get(0).getV.toString)
+    }.sum.setScale(currency.getDefaultFractionDigits, RoundingMode.HALF_EVEN)
+    val creditsRollup = rows.map { row =>
+      BigDecimal(row.getF.get(1).getV.toString)
+    }.sum.setScale(currency.getDefaultFractionDigits, RoundingMode.HALF_EVEN)
+
+    val spendSummary = SpendReportingForDateRange(
+      costRollup.toString(),
+      creditsRollup.toString(),
+      currency.getCurrencyCode,
+      startTime,
+      endTime
+    )
+
+    SpendReportingResults(Seq(dailySpendAggregation), spendSummary)
+  }
+
+  private def dateTimeToISODateString(dt: DateTime): String = dt.toString(ISODateTimeFormat.date())
+
+  private def getSpendExportConfiguration(billingProjectName: RawlsBillingProjectName): Future[BillingProjectSpendExport] = {
+    dataSource.inTransaction { dataAccess =>
+       dataAccess.rawlsBillingProjectQuery.load(billingProjectName)
+    }.map {
+      case Some(RawlsBillingProject(_, _, Some(billingAccount), _, _, _, _, _, Some(spendReportDataset), Some(spendReportTable), Some(spendReportDatasetGoogleProject))) =>
+        BillingProjectSpendExport(billingProjectName, billingAccount, Option(s"${spendReportDatasetGoogleProject.value}.${spendReportDataset.value}.${spendReportTable.value}"))
+      case Some(RawlsBillingProject(_, _, Some(billingAccount), _, _, _, _, _, _, _, _)) => BillingProjectSpendExport(billingProjectName, billingAccount, None)
+      case None => throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"project ${billingProjectName.value} not found"))
+      case _ => throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"billing account not found on billing project ${billingProjectName.value}"))
+    }
+  }
+
+  private def getWorkspaceGoogleProjects(billingProjectName: RawlsBillingProjectName): Future[Set[GoogleProject]] = {
+    dataSource.inTransaction { dataAccess =>
+        dataAccess.workspaceQuery.listWithBillingProject(billingProjectName)
+    }.map { workspaces =>
+      workspaces.collect {
+        case workspace if workspace.workspaceVersion == WorkspaceVersions.V2 => GoogleProject(workspace.googleProjectId.value)
+      }.toSet
+    }
+  }
+
+  private def validateReportParameters(startDate: DateTime, endDate: DateTime): Unit = {
+    if (startDate.isAfter(endDate)) throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "start date must be before end date"))
+    else if (Days.daysBetween(startDate, endDate).getDays > 90) throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "provided dates exceed maximum report date range"))
```

Comment:
It'd be good to extract the 90 magic number to a config value or a constant somewhere as this is very likely to change.

Also, same comment about including the parameter information in the error report msg to ease debugging.